From fc22448098f1638b97e8ed05053fb755220e0b12 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Wed, 05 Feb 2020 10:36:12 +0200
Subject: [PATCH] [SW-8229] flash: image selection in second chunk

Fix implementation of flash image selection to fit specs.
To select which is the main image we read the second chunk
of the fw, there we can find the pointer to which image to
use.

Change-Id: I267a76e86a59db2d91e44e6be668bbeb478e9963
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 586909c..77c850a 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -106,15 +106,10 @@
 static uint32_t arcboot_flash_image_offset(uint8_t image_ptr)
 {
 	uint32_t flash_image_offset;
-	uint32_t flash_image_switch_offset;
 	uint8_t flash_image_switch;
 
-	/* read offset in flash to primary/secondary image selection */
-	flash_image_switch_offset =
-		spi_flash_readl(FLASH_IMAGE_SWITCH_OFFSET);
-
 	/* read if we need to switch order of primary/secondary image */
-	flash_image_switch = spi_flash_readl(flash_image_switch_offset);
+	flash_image_switch = spi_flash_readl(FLASH_IMAGE_SWITCH_OFFSET);
 	if (flash_image_switch != 0) {
 		if (image_ptr != ARCBOOT_SECONDARY)
 			image_ptr = ARCBOOT_SECONDARY;
diff --git a/common/spi_flash.h b/common/spi_flash.h
index 5f1e690..95848a8 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -45,7 +45,7 @@
 						sizeof(uint32_t))
 #define FLASH_EFUSE_OFFSET		(FLASH_SECONDARY_OFFSET + \
 						sizeof(uint32_t))
-#define FLASH_IMAGE_SWITCH_OFFSET	(FLASH_EFUSE_OFFSET + sizeof(uint32_t))
+#define FLASH_IMAGE_SWITCH_OFFSET	4096
 
 #define FLASH_EFUSE_FW_VERSION_OFFSET	0
 #define FLASH_EFUSE_KEYS_OFFSET		16
